client/src/components/NewsFeed.js

```diff
@@ -0,0 +1,105 @@
+import React, { Component } from 'react';
+import { Segment, Feed, Image, Header, Icon, Divider, Modal, Grid} from 'semantic-ui-react'
+import { makeCall } from '../apis';
+import AlumniProfile from './AlumniProfile'
+
+/*
+ * DETAILS:
+ * NewsFeed component for visual display of news items created on backend
+ * Each type of post will recieve its own create(EVENT TYPE)Post method,
+ * called by constructDisplay
+ * PROPS
+ * userDetails - full profile of logged in user
+ * role - used to determine which types of news will be pulled
+ */
+export default class NewsFeed extends Component {
+    state={
+        newsItems: [],
+        display: []
+    }
+
+    async componentWillMount() {
+        let newsItems = await this.getNews(this.props.userRole)
+        await this.setState({
+            newsItems: newsItems.news.reverse()
+        })
+        this.constructDisplay()
+    }
+
+    getNews(role) {
+        return makeCall({}, '/news/getNews/'+{role}, 'get')
+    }
+
+    constructDisplay() {
+        let display = []
+        let i = 0
+        for (let feedItem of this.state.newsItems) {
+            if (feedItem.event === 'New Alumni') {
+                display.push(<Divider key={i}/>)
+                display.push(this.createNewAlumniPost(feedItem))
+            }
```

Comment:
For each new event type, we just need to add a conditional here

Reply:
Nice! Since there are going to be several types of newsItems, let's set this up as a `switch/case`